### DISCUSSION TITLE
Show CSS colors inline

### DISCUSSION THREAD

**Discussion by jeanas** [helix-editor/helix#11375] [Ideas]
Emacs has this cute little feature:

![image](https://github.com/user-attachments/assets/f1c87b9e-215d-45c6-b3d7-aa1bb4c641e7)

Does it sound feasible for Helix?

**Comment by David-Else** [comment]
I assume so, Helix uses Crossterm for cross platform terminal support, if the terminal supports 24bit color I don't see why not using RGB. Here is the API: https://docs.rs/crossterm/latest/crossterm/style/enum.Color.html#variant.Rgb

1. **iTerm2** - macOS terminal emulator that supports 24-bit color.
2. **GNOME Terminal** - Default terminal in many Linux distributions using GNOME, supports 24-bit color.
3. **Konsole** - KDE's terminal emulator supports 24-bit color.
4. **Alacritty** - A cross-platform, GPU-accelerated terminal emulator with 24-bit color support.
5. **xterm** - The classic terminal emulator for the X Window System has had 24-bit color support since version 327.
6. **kitty** - A feature-rich, fast, GPU-based terminal emulator that supports 24-bit color.
7. **Terminator** - Another Linux terminal emulator which supports 24-bit color.
8. **WezTerm** - A terminal emulator built with Rust, supporting 24-bit color and many other modern features.
9. **Windows Terminal** - The modern terminal for Windows 10 and later supports 24-bit color.


Run this to check your terminal:

```sh
awk 'BEGIN{
    s="/\\/\\/\\/\\/\\"; s=s s s s s s s s;
    for (colnum = 0; colnum<77; colnum++) {
        r = 255-(colnum*255/76);
        g = (colnum*510/76);
        b = (colnum*255/76);
        if (g>255) g = 510-g;
        printf "\033[48;2;%d;%d;%dm\033[38;2;%d;%d;%dm%s\033[0m", r,g,b, r,g,b, substr(s,colnum+1,1);
    }
    printf "\n";
}'
```

**Reply by jeanas** [reply]
I have syntax highlighting working in Helix (in GNOME Terminal), that's not the problem. The feature request here is for special syntax highlighting where the CSS color doesn't get highlighted just with the color configured for that token type, but actually gets the color that it represents.

**Reply by David-Else** [reply]
I understand the request, I was trying to show that it is possible to create a PR for it based on what Crossterm can do, and to show how to test if your terminal could display all those colors if Helix added it. I don't know how to program it, but I think it could be done, your question was `Does it sound feasible for Helix`, and my answer is yes... but I can't be sure as it would need to be only in CSS files, so probably it would be suitable for a plugin in the future.

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/issues/1115

**Reply by jeanas** [reply]
Ah, thanks, I missed that issue.

**Reply by TornaxO7** [reply]
@jeanas may I ask if you can mark your question as answered if it answered your question? :>

**Comment by nik-rev** [comment]
This is being worked on in https://github.com/helix-editor/helix/pull/12308
